restartmanager/restartmanager.go

```diff
@@ -72,11 +73,13 @@ func (rm *restartManager) ShouldRestart(exitCode uint32) (bool, chan error, erro
 
 	var restart bool
 	switch {
-	case rm.policy.IsAlways(), rm.policy.IsUnlessStopped():
+	case rm.policy.IsAlways():
+		restart = true
+	case rm.policy.IsUnlessStopped() && !hasBeenManuallyStopped:
 		restart = true
 	case rm.policy.IsOnFailure():
 		// the default value of 0 for MaximumRetryCount means that we will not enforce a maximum count
-		if max := rm.policy.MaximumRetryCount; max == 0 || rm.failureCount <= max {
+		if max := rm.policy.MaximumRetryCount; max == 0 || rm.restartCount < max {
```

Comment:
I replaced the "failureCount" in `restartManager` with `restartCount` for reusing the `restartManager.ShouldRestart()` in `container.ShouldRestart()`, when using failureCount, we're doing this:
1. failureCount++
2. Judge if failureCount <= max
After replaced with `restartCount`, we're doing:
1. Judge if restartCount < max
2. if need restart, restartCount++

And they have different meaning, `restartCount`means how many times container has been restarted before, it must be less than MaximumRetryCount, if the limit is reached, we shall not restart it again.